### DISCUSSION TITLE
Multiple install hooks

### DISCUSSION THREAD

**Discussion by pierreolivierh-rtone** [rauc/rauc#1556] [Q&A] [answered]
Hello,

Currently using Rauc on Yocto Kirkstone (v1.10.1), we successfully managed to install a hook to modify our bootloader (barebox), but I would like to add another hook script.

Currently, the bundle manifest looks like 
```
...
[hooks]
filename=barebox-updater.sh

[image.rootfs]
filename=image.tar.gz
hooks=post-install
```

Everything works as expected (the bundle is created, the hook is called).
I would like to add another hook (another script), and I used this documentation https://github.com/rauc/rauc/blob/v1.10.1/docs/using.rst#bundle-based-customization-hooks .

I tried to add another hook as stated in the doc, using `;` between hooks.
```
...
[hooks]
filename=barebox-updater.sh; test-updater.sh

[image.rootfs]
filename=yuze-g2-image-revpi-cm3.tar.gz
hooks=post-install;post-install
```
I had to modify a bit the bbclass to accept the `RAUC_BUNDLE_HOOKS[file] ?= "barebox-updater.sh;test-updater.sh"`

But I get an error from rauc when generating the bundle `Failed to create bundle: hook file 'barebox-updater.sh; test-updater.sh' does not exist in bundle content dir ` (both files are present in the bundle folder)

What did I miss ? Is it related to the rauc version ? 

Thanks a lot for any help
Pierre-Olivier

**Comment by ejoerns** [comment, marked as answer]
The documentation is maybe a bit misleading here. It actually says

> Multiple hook types must be separated with a `;`

This refers to the `hooks=` setting, not to the `filename=`. You can actually only have a single file in `filename` at the moment. (But this could call other files of course if required.)
